cache.rst

```diff
@@ -0,0 +1,192 @@
+
+    use Symfony\Contracts\Cache\ItemInterface;
+
+    // The callable will only be executed on a cache miss.
+    $value = $pool->get('my_cache_key', function (ItemInterface $item) {
+        $item->expiresAfter(3600);
+
+        // Do some HTTP request or heavy computations
+        $computedValue = 'foobar';
+
+        return $computedValue;
+    });
+
+    echo $value; // 'foobar'
+
+    $pool->delete('some_key');
+
+
+Symfony also supports PSR-6 and PSR-16 cache interfaces. You can read more about
+these at the :ref:`component documentation <component-cache>`.
+
+
+Configuring Cache with SymfonyFrameworkBundle
+---------------------------------------------
+
+When configuring the the cache component there are a few concepts you should know
+of:
+
+**Pool**
+    This is a service that you will interact with. Each pool will always have
+    its own namespace and cache items. There is never a conflict between pools.
+**Adapter**
+    An adapter is a *template* that you use to create Pools.
+**Provider**
+    A provider is the DSN connection to the actual storage.
+
+There are two pools that are always enabled by default. They are ``cache.app`` and
+``cache.system``. The system cache is use for things like annotations, serializer,
+and validation. The ```cache.app`` is used in your code. You can configure which
+adapter (template) they use by using the ``app`` and ``system`` key like:
+
+.. code-block:: yaml
+
+    framework:
+        cache:
+            app: cache.adapter.filesystem
+            system: cache.adapter.system
+
+The Cache component comes with a series of adapters already created:
+
+- :ref:`cache.adapter.apcu <component-cache-apcu-adapter>`
+- :ref:`ChainAdapter <component-cache-chain-adapter>`
+- :ref:`cache.adapter.doctrine <component-cache-doctrine-adapter>`
+- :ref:`cache.adapter.filesystem <component-cache-filesystem-adapter>`
+- :ref:`cache.adapter.memcached <component-cache-memcached-adapter>`
+- :ref:`cache.adapter.pdo <component-cache-pdo-doctrine-dbal-adapter>`
+- :ref:`PHPArrayAdapter <component-cache-php-array-adapter>`
+- :ref:`PHPFileAdapter <component-cache-php-files-adapter>`
+- :ref:`ProxyAdapter <component-cache-proxy-adapter>`
+- :ref:`cache.adapter.redis <component-cache-redis-adapter>`
+- cache.adapter.psr6
+- cache.adapter.system
+- NullAdapter
+- cache.adapter.array
+
+Some of these adapters could be configured via shortcuts:
+
+.. code-block:: yaml
+
+    framework:
+        cache:
+            directory: '%kernel.cache_dir%/pools' # Only used with cache.adapter.filesystem
+
+            default_doctrine_provider: 'app.doctrine_cache'
+            default_psr6_provider: 'app.my_psr6_service'
+            default_redis_provider: 'redis://localhost'
+            default_memcached_provider: 'memcached://localhost'
+            default_pdo_provider: 'doctrine.dbal.default_connection'
+
+Using the *providers* above, they will configure the adapters and create pools (services)
+with service ids:
```

Comment:
```suggestion
with these service ids:
```

Reply:
I think we should include this list in the code example directly, like:

```yaml
# service: cache.doctrine
default_doctrine_provider: 'app.doctrine_cache'
```

In the past, we've learnt that developers often learn more from comments in code examples than from the text around it :)

Reply:
Good idea!